iothub/device/src/Transport/AmqpIot/AmqpIotLinkFactory.cs

```diff
@@ -6,48 +6,43 @@
 using Microsoft.Azure.Amqp;
 using Microsoft.Azure.Devices.Shared;
 
-namespace Microsoft.Azure.Devices.Client.Transport.AmqpIoT
+namespace Microsoft.Azure.Devices.Client.Transport.AmqpIot
 {
-    internal class AmqpIoTLinkFactory : ILinkFactory
+    internal class AmqpIotLinkFactory : ILinkFactory
     {
-        private static AmqpIoTLinkFactory s_instance = new AmqpIoTLinkFactory();
+        private static readonly AmqpIotLinkFactory s_instance = new AmqpIotLinkFactory();
 
-        private AmqpIoTLinkFactory()
+        private AmqpIotLinkFactory()
         {
             // Important: must not throw as it's used within the static ctor.
         }
 
-        public static AmqpIoTLinkFactory GetInstance()
+        public static AmqpIotLinkFactory GetInstance()
         {
             return s_instance;
         }
 
         public IAsyncResult BeginOpenLink(AmqpLink link, TimeSpan timeout, AsyncCallback callback, object state)
         {
-            Debug.Fail($"{nameof(AmqpIoTLinkFactory)} open link should not be used.");
+            Debug.Fail($"{nameof(AmqpIotLinkFactory)} open link should not be used.");
             throw new NotImplementedException();
         }
 
         public AmqpLink CreateLink(AmqpSession session, AmqpLinkSettings settings)
         {
             if (Logging.IsEnabled)
             {
-                Logging.Info(this, session, $"{nameof(CreateLink)}");
+                Logging.Info(this, session, nameof(CreateLink));
             }
 
-            if (settings.IsReceiver())
-            {
-                return new ReceivingAmqpLink(session, settings);
-            }
-            else
-            {
-                return new SendingAmqpLink(session, settings);
-            }
+            return settings.IsReceiver()
+                ? new ReceivingAmqpLink(session, settings)
+                : (AmqpLink)new SendingAmqpLink(session, settings);
```

Comment:
q - why did only one of the code paths require an explicit cast?

Reply:
Good question. Removing it causes this error.

> error CS8400: Feature 'target-typed conditional expression' is not available in C# 8.0. Please use language version 9.0 or greater.

Either can have the cast to make the issue go away. If both have it, the IDE thinks one is unnecessary.

Weird, right?